omod/src/main/java/org/openmrs/module/pihcore/page/controller/visit/TodaysVisitsPageController.java

```diff
@@ -37,12 +37,12 @@ public void get(PageModel model,
 
         Location location = uiSessionContext.getSessionLocation();
 
-        if(!location.hasTag(visitLocationTag.toString())){
+        while(location != null && !location.hasTag(visitLocationTag.toString())){
```

Comment:
Looks good, thanks @louidorjp !

One small formatting nitpick, by convention we put space between a command a paranthesis and a bracket... (but we can just handle this in a post-commit change, I'll merge this in!